tests/monitoring/metrics.go

```diff
@@ -0,0 +1,150 @@
+package monitoring
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/onsi/gomega/types"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	virtapi "kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-api"
+	virtcontroller "kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-controller"
+	virtoperator "kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-operator"
+
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libmonitoring"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[Serial][sig-monitoring]Metrics", Serial, decorators.SigMonitoring, func() {
+	var virtClient kubecli.KubevirtClient
+	var metrics *libmonitoring.QueryRequestResult
+	var err error
+
+	setupVM := func() {
+		vmi := libvmi.NewCirros(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		)
+		vmi.Namespace = testsuite.GetTestNamespace(nil)
+
+		By("Creating a running VirtualMachine")
+		vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunning())
+		_, err = virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() error {
+			vmi, err = kubevirt.Client().VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+			return err
+		}, 120*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		libwait.WaitForSuccessfulVMIStart(vmi)
+
+		libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", 1)
+
+		By("Deleting the VirtualMachine")
+		err = virtClient.VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", -1)
+	}
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+
+		setupVM()
```

Comment:
This is a expensive operation, to spin up a VM and delete it on each test spec, can you please explain why do you need to do this check especially when you are testing control plane component metrics?

Reply:
we test for the presence of all KubeVirt metrics, including, for example, VM stats metrics, which need a VM to exist to be created

Reply:
and this should be a beforeall, we only need to run this once

Reply:
Why beforeAll? Can't you just test all the relevant VMI metrics in one test spec?

Reply:
is it not better organized splitted between multiple `It`s?

Reply:
From [Ginkgo](https://onsi.github.io/ginkgo/#setup-in-ordered-containers-beforeall-and-afterall) docs I understand the tests need to include `Ordered` label, AFAIK its usage was discouraged in our e2e suites.

Reply:
removed